src/components/organisms/AdminVenueView/components/Container/Container.tsx

```diff
@@ -0,0 +1,317 @@
+import React, {
+  useCallback,
+  useState,
+  useMemo,
+  CSSProperties,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+} from "react";
+import { useDrop } from "react-dnd";
+import update from "immutability-helper";
+import ReactResizeDetector from "react-resize-detector";
+
+import { Dimensions, Position } from "types/utility";
+import { RoomData_v2 } from "types/rooms";
+
+import { DragItem } from "pages/Account/Venue/VenueMapEdition/interfaces";
+import { ItemTypes } from "pages/Account/Venue/VenueMapEdition/ItemTypes";
+import { DraggableSubvenue } from "pages/Account/Venue/VenueMapEdition/DraggableSubvenue";
+import { CustomDragLayer } from "pages/Account/Venue/VenueMapEdition";
+import { snapToGrid as doSnapToGrid } from "pages/Account/Venue/VenueMapEdition/snapToGrid";
+
+const styles: React.CSSProperties = {
+  width: "100%",
+  position: "relative",
+};
+
+export interface RoomIcon {
+  top: number;
+  left: number;
+  url: string;
+  width: number;
+  height: number;
+  roomIndex?: number;
+}
+
+export interface RoomIconsMap {
+  [key: string]: RoomIcon;
+}
+
+interface CoordinatesBoundary {
+  width: number;
+  height: number;
+}
+
+interface PropsType {
+  snapToGrid?: boolean;
+  roomIcons?: RoomIcon[];
+  backgroundImage: string;
+  iconImageStyle?: CSSProperties; // This is not being used ATM
+  draggableIconImageStyle?: CSSProperties; // This is not being used ATM
+  roomIconsMap?: RoomIconsMap;
+  onOtherIconClick?: (key: string) => void;
+  coordinatesBoundary: CoordinatesBoundary;
+  interactive: boolean;
+  resizable: boolean;
+  onResize?: (size: Dimensions) => void;
+  onMove?: (position: Position) => void;
+  otherIconsStyle?: CSSProperties;
+  rounded?: boolean;
+  backgroundImageStyle?: CSSProperties;
+  containerStyle?: CSSProperties;
+  lockAspectRatio?: boolean;
+  rooms: RoomData_v2[];
+  selectedRoom: RoomData_v2 | undefined;
+  setSelectedRoom: Dispatch<SetStateAction<RoomData_v2 | undefined>>;
+}
+
+export const Container: React.FC<PropsType> = ({
```

Comment:
The name of the component is too generic tbh.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553